### DISCUSSION TITLE
What is the difference between addKeybindingRule and addCommand?

### DISCUSSION THREAD

**Discussion by chl3188** [microsoft/monaco-editor#3888] [Q&A] [answered]
I think the difference is that the 'when' is added to the addKeybindingRule. 
But I don't understand exactly what two functions are used in.

Can you explain it with an example?

**Comment by tamayika** [comment, marked as answer]
addKeybindingRule can add keybinding to existing command which exists in editor by default.
Please see [example](https://microsoft.github.io/monaco-editor/playground.html?source=v0.37.1#example-interacting-with-the-editor-adding-a-keybinding-to-an-existing-command).
